backend/server.js

```diff
@@ -195,30 +247,46 @@ sbp('sbp/selectors/register', {
       // This is done in a queue to handle several simultaneous requests
       // reading and writing to the same key
       await appendToIndexFactory(resourcesKey)(resourceID)
+      sbp('chelonia.persistentActions/enqueue', ['backend/server/saveIndirectResourcesIndex', resourceID])
```

Comment:
Why is this done as a persistent action? It seems rather quick and short-lived. Should be safe to just do it here, no?

(If you disagree, could you add a comment to the code explaining?)

Reply:
Since this is a recursive task, it seemed better to do it later. It may be short lived, but there can be contention on a contract that has lots of indirect resources, as the index needs to be locked.